packages/php-wasm/web-service-worker/src/initialize-service-worker.ts

```diff
@@ -13,6 +15,14 @@ import { getURLScope, isURLScoped, setURLScope } from '@php-wasm/scopes';
 export function initializeServiceWorker(config: ServiceWorkerConfiguration) {
 	const { handleRequest = defaultRequestHandler } = config;
 
+	self.addEventListener('install', (event) => {
+		try {
+			event.waitUntil(preCacheResources());
```

Comment:
I saw line targets `index.html`, `sw.js`, and `/` – why those files specifically and not also, say, `style.css`? Also, using `waitUntil` means the service worker won't start serving other requests until those files are cached – was that intended?

Reply:
If we don't use `event.waitUntil` we can't ensure all the files will be available for offline mode after the initial load, but I don't think this is a big problem and in my testing, it works without `event.waitUntil`.

Reply:
> I saw line targets index.html, sw.js, and / – why those files specifically and not also, say, style.css?

This is the root cause of _There is still an issue on the first load where some static assets aren't pulled from cache._. All files should be preached for offline support to work after the first load. 

I'm searching for a solution now.

Reply:
This is working now. I had to preload all assets during `install` to ensure they are available while offline.

I'm not happy with the implementation, it requires two fetches to get the asset list, and prel all assets.

On one had this is more robust and allows us to do things like change PHP versions while offline, but I would still like to reduce the list size.